### PR TITLE
fix: `ScriptResultDecoderError` on dry run calls

### PR DESCRIPTION
**Pull request by Torres-ssf**
- Close https://github.com/FuelLabs/fuels-ts/issues/1245

### CODE REVIEW

File: packages/fuel-gauge/src/contract.test.ts [hunk 1]
```diff
@@ -946,4 +946,24 @@ describe('Contract', () => {
 
     expect(finalBalance).toBe(initialBalance + amountToContract);
   });
+
+  it('should ensure ScriptResultDecoderError works for dryRun and simulate calls', async () => {
```
Comment (danielbate) [inline packages/fuel-gauge/src/contract.test.ts:950]: Says `simulate` but we only seem to `dryRun` in the test case. I appreciate these are coupled at the moment but they could be decoupled at some point, is it worth having a separate test case for `simulate`?

Comment (danielbate) [inline packages/fuel-gauge/src/contract.test.ts:950]: Says `simulate` but we only seem to `dryRun` in the test case. I appreciate these are coupled at the moment but they could be decoupled at some point, is it worth having a separate test case for `simulate`?

Comment (Torres-ssf) [reply packages/fuel-gauge/src/contract.test.ts:950]: I've just improved the test case. But at the end of the day both `dryRun` and `simulate` will run the `operation.dryRun`.

**Review by nedsalk — APPROVED** [review]

**Review by arboleya — APPROVED** [review]
Awesome!

**Pull request merged**